distributed/core.py

```diff
@@ -61,32 +61,6 @@ class Status(Enum):
     undefined = None
     dont_reply = "dont-reply"
 
-    def __eq__(self, other):
-        """
-        Implement equality checking with backward compatibility.
-
-        If other object instance is string, we compare with the values, but we
-        actually want to make sure the value compared with is in the list of
-        possible Status, this avoid comparison with non-existing status.
-        """
```

Comment:
Given the need for some backwards compatibility, I wonder if we might first try the following:

```python
try:
    return self.value == other.value
except AttributeError:
    ...
```

My experience is that try-except tends to be quite a bit faster than if/else in the try case, and this avoids the type check as well (which should be fine).

We can also migrate over all of the other projects if that will provide a large boost.  It isn't hard to do.